src/OpenToolkit.Mathematics/Color/Color3.cs

```diff
@@ -0,0 +1,1231 @@
+        /// Compares the specified Color4 structures for equality.
+        /// </summary>
+        /// <param name="left">The left-hand side of the comparison.</param>
+        /// <param name="right">The right-hand side of the comparison.</param>
+        /// <returns>True if left is equal to right; false otherwise.</returns>
+        [Pure]
+        public static bool operator ==(Color3<T> left, Color3<T> right)
+        {
+            return left.Equals(right);
+        }
+
+        /// <summary>
+        /// Compares the specified Color4 structures for inequality.
+        /// </summary>
+        /// <param name="left">The left-hand side of the comparison.</param>
+        /// <param name="right">The right-hand side of the comparison.</param>
+        /// <returns>True if left is not equal to right; false otherwise.</returns>
+        [Pure]
+        public static bool operator !=(Color3<T> left, Color3<T> right)
+        {
+            return !left.Equals(right);
+        }
+
+        /// <summary>
+        /// Compares whether this Color4 structure is equal to the specified Color4.
+        /// </summary>
+        /// <param name="other">The Color4 structure to compare to.</param>
+        /// <returns>True if both Color4 structures contain the same components; false otherwise.</returns>
+        [Pure]
+        public bool Equals(Color3<T> other)
+        {
+            return X.Equals(other.X) && Y.Equals(other.Y) && Z.Equals(other.Z);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="string"/> that describes this Color4 structure.
+        /// </summary>
+        /// <returns>A <see cref="string"/> that describes this Color4 structure.</returns>
+        [Pure]
+        public override string ToString()
+        {
+            return $"({X}, {Y}, {Z})";
+        }
+
+        /// <summary>
+        /// Compares whether this Color4 structure is equal to the specified object.
+        /// </summary>
+        /// <param name="obj">An object to compare to.</param>
+        /// <returns>True obj is a Color4 structure with the same components as this Color4; false otherwise.</returns>
+        [Pure]
+        public override bool Equals(object obj)
+        {
+            return obj is Color3<T> other && Equals(other);
+        }
+
+        /// <summary>
+        /// Calculates the hash code for this Color4 structure.
+        /// </summary>
+        /// <returns>A System.Int32 containing the hashcode of this Color4 structure.</returns>
+        [Pure]
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = X.GetHashCode();
+                hashCode = (hashCode * 397) ^ Y.GetHashCode();
+                hashCode = (hashCode * 397) ^ Z.GetHashCode();
+                return hashCode;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Factory for conversion between color types.
+    /// </summary>
+    public static class Color3
```

Comment:
Id also like to move Color*\<T\> out into its own file... ``Color*`T.cs`` is a common pattern.

Reply:
I would like @varon to give an opinion on this, as i somewhat agree here. What i would probably do is moving each color space into their own file having the extension methods in there aswell. Example
One file is Hsv.cs, all the extension methods converting X->Hsv is in that file, plus any extra extensions